### PR TITLE
Oracle index

### PR DESCRIPTION
**Pull request by DrPresident**
Initial testing looks good, I can test more thoroughly once we have BAND on pulsar-1. Currently uses a basic weighted average
```
price | weight
1.00  | .5
7.5    | .5
sum((price * weight), ...) / total_weight = price
((1.0 * .5) + (7.5 * .5)) / (.5 + .5) = 4.25
```
```
price | weight
1.00  | .2
7.5    | .5
sum((price * weight), ...) / total_weight = price
((1.0 * .2) + (7.5 * .5)) / (.2 + .5) = 5.642857142857143
```
I used this site to build test cases https://www.rapidtables.com/calc/math/weighted-average-calculator.html
This still needs testing with sswap pairs and such, but using band it is working well

### CODE REVIEW

File: contracts/oracle/README.md [hunk 1]
```diff
@@ -14,7 +14,8 @@ The oracle contract is used to query the price of different currencies
 ##### Request
 |Name      |Type      |Description                                                                                                        | optional |
 |----------|----------|-------------------------------------------------------------------------------------------------------------------|----------|
-|owner     | string   |  New contract owner; SHOULD be a valid bech32 address, but contracts may use a different naming scheme as well    |  yes     |
+|admin     | string   |  New contract owner; SHOULD be a valid bech32 address, but contracts may use a different naming scheme as well    |  yes     |
```
Comment (FloppyDisck) [inline contracts/oracle/README.md:17]:
```suggestion
|admin     | string   |  New contract admin; SHOULD be a valid bech32 address, but contracts may use a different naming scheme as well    |  yes     |
```

File: test-index.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
```
Comment (FloppyDisck) [inline test-index.py:1]: Since the indexes are self contained equations, could you make unit tests in addition to these integration tests?

**Review by FloppyDisck — CHANGES_REQUESTED** [review]

File: test-index.py (hunk 2 above)
Comment (DrPresident) [reply test-index.py:1]: Basically all the functions require queries to either band or secretswap

Comment (FloppyDisck) [reply test-index.py:1]: Could those functions be moved into individual functions so you can spoof that data?

**Review by FloppyDisck — APPROVED** [review]

**Pull request merged**